File: hh-front/src/models/conductor_model.js
import { UsuarioModel } from './usuario_model'
import { CamionModel } from './camion_model'
import { LicenciaModel } from './licencia_model'

export class ConductorModel {
  constructor () {
    this.id = ''
    this.adelantos = new Array([])
    this.camion = new CamionModel()
    this.licencia = new LicenciaModel()
    this.direccion = ''
    this.email = ''
    this.identificacion = ''
    this.nombre = ''
    this.notas = ''
    this.telefono = ''

    this.creador = new UsuarioModel()
    this.creada = ''
    this.modificador = new UsuarioModel()
    this.modificada = ''
    this.eliminador = new UsuarioModel()
    this.elimianda = ''
  }
}

File: hh-back/src/main/java/gloit/hiperionida/helios/model/ConductorModel.java
package gloit.hiperionida.helios.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = false)
@Entity
@Getter
@Setter
@SuperBuilder
@Table(name = "conductor")
public class ConductorModel extends AbsPersonaModel {
    public ConductorModel() {
    }

    public ConductorModel(Long id, Long creadorId, LocalDateTime creada, Long modificadorId, LocalDateTime modificada, Long eliminadorId, LocalDateTime eliminada, String direccion, String email, String identificacion, String nombre, String notas, String telefono) {
        super(id, creadorId, creada, modificadorId, modificada, eliminadorId, eliminada, direccion, email, identificacion, nombre, notas, telefono);
    }
}

